### DISCUSSION TITLE
Support SQL Server shared memory protocol?

### DISCUSSION THREAD

**Discussion by suimong** [sfu-db/connector-x#169] [Ideas]
Hi,

I wonder if `connector-x` support connecting to SQL Server using the shared memory protocol. I'm encountering a use case where the task is to export large sql server tables to parquet files which can be done on-premise. So theoretically it would be faster to transfer data using shared memory protocol instead of tcp/ip. Is this workable or already supported? 

Thanks for making this wonderful library.

**Comment by wangxiaoying** [comment]
Hi @suimong, we do not support shared memory protocol right now. Underlying, we use [Tiberius](https://github.com/prisma/tiberius) as client driver which is independent from the underlying protocol, and according to [this](https://github.com/prisma/tiberius#supported-protocols) they didn't find a proper crate to support shared memory protocol. So unfortunately I'm afraid it cannot be supported in a short time. But I will follow up this issue. Thanks for the suggestion!

**Reply by suimong** [reply]
Thanks, appreciated!
